app/controllers/application_controller.rb

```diff
@@ -254,6 +254,13 @@ def load_users
     @users = User.accessible_by(current_ability).order(name: :asc)
   end
 
+  def set_current_step
+    return unless current_intake.present?
+    return unless request.method_symbol == :get # skip uploads
+
+    current_intake.update(current_step: current_path) unless current_intake.current_step == current_path
+  end
```

Comment:
I attempted to test these in the documents controller spec + questions controller spec, but take a look at those specs -- they aren't even actual controller specs, and when I attempted to convert them into controller specs I ran into issues with them because the routes are set up in some funky ways.